examples/tut-6-rpc/client-py/index.py

```diff
@@ -0,0 +1,108 @@
+import json
+import os
+import random
+import requests
+from concurrent.futures import ALL_COMPLETED, ThreadPoolExecutor, wait
+from time               import sleep
+
+
+def log(t):
+    print(t, flush=True)
+
+
+def mk_rpc(proto, host, port):
+    def rpc(m, *args):
+        lab = 'RPC %s %s' % (m, json.dumps([*args]))
+        log(lab)
+        ans = requests.post('%s://%s:%s%s' % (proto, host, port, m),
+                            json=[*args])
+        log('%s ==> %s' % (lab, json.dumps(ans.json())))
+        return ans.json()
+
+    def rpc_callbacks(m, arg, cbacks):
+        vals  = {k: v    for k, v in cbacks.items() if not callable(v)}
+        meths = {k: True for k, v in cbacks.items() if     callable(v)}  # noqa
+        p     = rpc(m, arg, vals, meths)
+
+        while True:
+            if p['t'] == 'Done':
+                return p
+
+            elif p['t'] == 'Kont':
+                cback = cbacks[p['m']]
+                ans   = cback(*p['args'])
+                p     = rpc('/kont', p['kid'], ans)
+
+            else:
+                # TODO fix swallowed exceptions
+                raise Exception('Illegal callback return: %s' % json.dumps(p))
+
+    return rpc, rpc_callbacks
+
+
+def main():
+    log('I am the client')
+    sleep(3)  # TODO wait until server becomes available to fulfill requests
+    host = os.environ['REACH_RPC_SERVER']
+    port = os.environ['REACH_RPC_PORT']
+
```

Comment:
Can you move this into `mk_rpc`... maybe use them if no args are provided?

Reply:
Moved in a98cd5e